### ISSUE TITLE
Issue after calling removeTag

### ISSUE THREAD

**Issue by nahun-accesium** [PHPSocialNetwork/phpfastcache#392]
I'm using `phpfastcache` in a proyect and I need to clean all the cached queries related to an entity. I'm trying to tag all those elements so I was playing with the `addTags`, `setTags` and `removeTag` methods.

After calling removeTag I can no longer save the Cache Item. It's worth noting that the error does not occur If I remove a non existing tag. Also `getItemsByTag` retrieves elements no longer tagged. Am I doing something wrong?

Here is a small test case to reproduce the issue:

```php
$InstanceCache = CacheManager::getInstance('memcached');
//$InstanceCache->clear();

$CachedElement1 = $InstanceCache->getItem('el1');
$CachedElement1->set('element1')->expiresAfter(600);
$CachedElement1->addTags(array('tag_12'));
$InstanceCache->save($CachedElement1);
//var_dump($CachedElement1->getTags()); Outputs: tag_12

$CachedElement1 = $InstanceCache->getItem('el1');
$CachedElement1->setTags(array('tag_34'));
$InstanceCache->save($CachedElement1);
//var_dump($CachedElement1->getTags()); Outputs: tag_34

$CachedElement1 = $InstanceCache->getItem('el1');
$CachedElement1->removeTag('tag_12');

// Save after removing a non existing tag: works as expected
try {
	$InstanceCache->save($CachedElement1);
} catch (Exception $e) {
	echo $e->getMessage();
}
//var_dump($CachedElement1->getTags()); Outputs: tag_34

$CachedElement1 = $InstanceCache->getItem('el1');
$CachedElement1->removeTags(array('tag_34'));

// Save after removing an existing tag: fails
try {
	$InstanceCache->save($CachedElement1);
} catch (Exception $e) {
	echo $e->getMessage();
	// Outputs: $key must be a string, got type "object" instead.
}
//var_dump($CachedElement1->getTags()); Outputs: empty

// Finally, fetch element by the original tag retrieves the item, despite the getTags method retrieving an empty set
$items = $InstanceCache->getItemsByTag('tag_12');

foreach ($items as $item) {
	var_dump($item->getKey());
	var_dump($item->get());
	var_dump($item->getTags());
}

// Outputs: string(3) "el1" string(8) "element1" array(0) { }
```

Environment:

HHVM 3.13.1
PHP 5.6.99-hhvm
memcached 2.2.0

**Comment by Geolim4** [comment]
Hello thanks for the test, I'm having a look at it :)

Cheers,
Georges

**Comment by Geolim4** [comment]
Bug identified, a patch is coming.
I will also add an anti-regression test to this specific case.

Thanks you,
Georges

**Comment by nahun-accesium** [comment]
Glad I helped. Good work on the library. Congratulations.

**Comment by Geolim4** [comment]
A test has been added:
https://github.com/PHPSocialNetwork/phpfastcache/commit/de48dc3b91799ab915785e103a9726a03b544dd7#diff-254397cb7e900c8d3550a79126c0256e

**Comment by Geolim4** [comment]
Merged on master via #393
Thanks you for your report :)
